spec/hoosegow_docker_spec.rb

```diff
@@ -64,6 +64,42 @@
     end
   end
 
+  context "with custom connection options" do
+    before(:each) {
+      FileUtils.remove_dir(test_dir, true)
+      FileUtils.mkpath(test_dir)
+    }
+    after { FileUtils.remove_dir(test_dir, true) }
+
+    let(:test_dir) { File.join(Dir.pwd, 'volume-test') }
+
+    it "configures ExtraHosts option" do
+      config = CONFIG.merge(
+        :Entrypoint => ['/bin/cp',  '/etc/hosts', '/volume-test/hosts'],
```

Comment:
> Is /etc/hosts used as an example file so that the test can verify that something comes out on the volume?

The test serves two purposes: that the volume bind configuration introduced in #17 is not broken (we can copy any file into the mounted volume and read it outside of the container) and that we have successfully configured the `:ExtraHosts` option in the `#create-a-container` docker API. Below is the description for this option for the [docker API](https://docs.docker.com/engine/api/v1.19/) under the `:HostConfig` keyI:

```
ExtraHosts - A list of hostnames/IP mappings to add to the container’s /etc/hosts file. Specified in the form ["hostname:IP"].
```

Another way to verify that the `:ExtraHosts` option is no longer being ignored and being properly set is to introduce a `container_info` method to directly check the running containers config:

```
    # Public: Get more information about the running container
    #
    # Returns response body or nil if no container is running.
    def container_info
      @container.json
    end
```

This would give us docker inspect functionality and we can easily check to see that any configuration we've set is present:

![image](https://user-images.githubusercontent.com/839181/62652743-ae400000-b921-11e9-9cbd-817cc245dd87.png)

> Can this bug be fixed without bringing in activesupport?

Definitely! Selectively merging into an existing `:HostConfig` option would be the way to do this. Let me try a few things and get something up shortly.

Reply:
> The test serves two purposes: that the volume bind configuration introduced in #17 is not broken (we can copy any file into the mounted volume and read it outside of the container) and that we have successfully configured the :ExtraHosts option in the #create-a-container docker API. 

Ah, word. I didn't realize there were two purposes. From a test purity perspective, it's better to check one thing per test. But for stuff like this sometimes it's easier to check just one thing.  👍 on using `/etc/hosts` if you'd like to keep just one test here.